### DISCUSSION TITLE
In citation_gat example Keras evaluate function gives different results from scikit learn accuracy_score

### DISCUSSION THREAD

**Discussion by PietroNardelli** [danielegrattarola/spektral#298] [Q&A] [answered]
Hello,

I am running the citacion_gat.py example but I realized that if I use the scikit learn accuracy_score function instead of the model.evaluate on the testing dataset I get different results. Is there a reason for this? Does it depend on how the data is extracted by the SingleLoader? 

Here is what I am doing in order to evaluate the method with sklearn: 

from sklearn.metrics import accuracy_score

data = loader_te.__next__()
feat = data[0][0]
adj_matrix = data[0][1]

y_pred = model.predict([feat, adj_matrix], batch_size=1)
y_pred = y_pred[weights_te>0]
y_true = data[1]
y_true = y_true[weights_te>0]

print('sklearn accuracy: {}'.format(accuracy_score(np.argmax(y_true, axis=1), np.argmax(y_pred, axis=1))))

eval_results = model.evaluate(loader_te.load(), steps=loader_te.steps_per_epoch)
print("Done.\n" "Test loss: {}\n" "Test accuracy: {}".format(*eval_results))

Results:
sklearn accuracy: 0.741
1/1 [==============================] - 0s 21ms/step - loss: 0.9705 - acc: 0.8320
Done.
Test loss: 0.9705386757850647
Test accuracy: 0.8320000171661377

**Comment by danielegrattarola** [comment, marked as answer]
Hi, 

if the data is from a SingleLoader, then this line:

```python
model.predict([feat, adj_matrix], batch_size=1)
```
should crash because it should have `batch_size=N`.
The fact that it doesn't crash tells me that something has gone horribly wrong in terms of silent bugs :D 

Can you confirm that you still have the issue if you set `batch_size=feat.shape[0]`?

Thanks

**Reply by PietroNardelli** [reply]
You are right, using batch_size=feat.shape[0] works! Now I am getting the same results with scikit learn.

Thanks a lot!
